### ISSUE TITLE
Add an option to only backup to Drive snapshots the addon creates.

### ISSUE THREAD

**Issue by sabeechen** [sabeechen/hassio-google-drive-backup#353]
EG If the option were enabled, then any snapshots you create through the Supervsior snapshot page or automations wouldn't get uploaded to Google Drive by the addon.  Default would be to upload everything (the current behavior).

**Comment by sabeechen** [comment]
Originally suggested by [Uzirox](https://community.home-assistant.io/t/add-on-home-assistant-google-drive-backup/107928/453?u=sabeechen) on the Home Assistant forums.

**Comment by vanstinator** [comment]
The latest releases of the Supervisor are now creating snapshots for every little addon update. So this feature would be super appreciated to keep that cruft out of drive.

**Comment by skippy-oz** [comment]
I would like this added as I have just done 6 addon updates so I no longer have any snapshots store in my google drive

**Comment by sabeechen** [comment]
I'll add this in soon (within a few weeks) as I've got a bunch of bugs I need to work through.  In the meantime I'd recommend turning off auto-update if you have limited space so you can update manually update addons and deselect the option to generate a snapshot too.

**Comment by sabeechen** [comment]
It took me a frustrating amount of time to figure out how to implement this well, but I've released this in v0.104.1 of the addon just now.

To update the addon, go to:
`Supervisor` > `Add-on Store` > Click the `...` in the top right > `Refresh`
and you should see the new version.

I ended up adding two new config options related to this bug:
  - ```ignore_other_snapshots: True``` will cause the add-on to ignore anything it didn't create.
  - ```ignore_upgrade_snapshots: True``` will cause the addon to ignore anything that "looks like" an upgrade snapshot.  This means any partial snapshot it didn't make that only has one folder or addon inside of it.

In addition the UI now has a special section below the list of snapshots where you can see all of the snapshots its ignoring and choose if you'd like to include them in backups individually.  The trickiness came in two parts:
 - What should the addon do with snapshots that _should_ be ignored but are already backup up to Google Drive?
 - How should the addon identify the snapshots it created?

What I ended up going with after spending waaaaaay too much time thinking through the different use cases is:
 - If a snapshot has been uploaded to Google Drive already, it won't be considered "ignored" until the snapshot in Google Drive is gone.
 - The addon will keep a small database of the snapshots its created to help it identify which snapshots its made.  Unfortunately there is no way the addon could know concretely if a snapshot made before installing the version that tracks them was made by itself or not.

Going with this strategy means that the addon will never incorrectly delete a snapshot (data loss is always my primary concern), but it _does_ mean that after you turn the new ignore feature on it might incorrectly assume that it didn't create a snapshot you made before installing the new version.  After you turn this feature on, I'd recommend looking through the list of snapshots its ignoring to make sure you don't actually want to back some of them up.  If you have an "upgrade" snapshot that isn't getting ignored, just delete it from Google Drive and it should start getting ignored.  This is the best compromise that I could come up with that doesn't lose data and doesn't inconvenience users too much.

Anyway this is kind of turning into a wall of text:
tl;dr
 - Turn on the new feature from the addon's setting dialog.
 - Look through your addons to make sure everything is getting ignored to your liking
   - If its getting ignored but should, click the "Don't ignore" button.
   - If its getting backed up but should, just delete the snapshot from Google Drive.
   - Reopen this bug and kindly let me know if you're still running into problems.

**Comment by vanstinator** [comment]
Thank you! This seems to work as described. And the new UI is also quite nice. Lovely QoL update. ❤️
